src/services/InterviewService.js

```diff
@@ -767,6 +773,82 @@ async function updateCompletedInterviews () {
  */
 async function partiallyUpdateInterviewByWebhook (interviewId, webhookBody) {
   logger.info({ component: 'InterviewService', context: 'partiallyUpdateInterviewByWebhook', message: `Received webhook for interview id "${interviewId}": ${JSON.stringify(webhookBody)}` })
+
+  // this method is used by the Nylas webhooks, so use M2M user
+  const m2mUser = helper.getAuditM2Muser()
+  const { page: pageDetails, booking: bookingDetails } = webhookBody
+  const interviewStartTimeMoment = moment.unix(bookingDetails.start_time)
+  const interviewEndTimeMoment = moment.unix(bookingDetails.end_time)
+  let updatedInterview
+
+  if (bookingDetails.is_confirmed) {
+    try {
+      // update the Nylas event to set custom metadata
+      const accountEmail = await getAccountEmail(bookingDetails.account_id)
+      if (!accountEmail) {
+        throw new errors.BadRequestError('Error getting account email for the given account id.')
+      }
+
+      const authorizationCode = await authenticateAccount(bookingDetails.account_id, accountEmail)
+      if (!authorizationCode) {
+        throw new errors.BadRequestError('Error getting account authorization code for the given account.')
+      }
+
+      const { accessToken } = await getAccessToken(authorizationCode)
+      if (!accessToken) {
+        throw new errors.BadRequestError('Error getting access token for the given account.')
+      }
```

Comment:
We don't have to request access token again from Nylas because we already have it stored in the DB. See how we do it inside update interview endpoint https://github.com/topcoder-platform/taas-apis/blob/dev/src/services/InterviewService.js#L408-L415

Reply:
Done in latest commit.